Not closing show_text quote leads to a crash

Steps to reproduce
- Type `<hover:show_text:'>` while holding a styled text
- crash!

stack trace
```
Caused by: java.lang.ClassCastException: class io.github.homchom.recode.ui.text.StyledString cannot be cast to class net.kyori.adventure.text.BuildableComponent (io.github.homchom.recode.ui.text.StyledString and net.kyori.adventure.text.BuildableComponent are in unnamed module of loader net.fabricmc.loader.impl.launch.knot.KnotClassLoader @6cc4c815)
    at io.github.homchom.recode.feature.visual.ExpressionHighlighter.highlightUncached$lambda$1(ExpressionHighlighting.kt:109) ~[recode-0.1.0-beta.10+1.20.4-LATEST.jar:?]
    at net.kyori.adventure.text.AbstractComponentBuilder.mapChildren(AbstractComponentBuilder.java:162) ~[net_kyori_adventure-api-4.15.0-dda99c57db8c2b82.jar:?]
    at io.github.homchom.recode.feature.visual.ExpressionHighlighter.highlightUncached(ExpressionHighlighting.kt:107) ~[recode-0.1.0-beta.10+1.20.4-LATEST.jar:?]
    at io.github.homchom.recode.feature.visual.ExpressionHighlighter.highlightString(ExpressionHighlighting.kt:74) ~[recode-0.1.0-beta.10+1.20.4-LATEST.jar:?]
    at io.github.homchom.recode.feature.visual.EditBoxExpressionFormatter.format(EditBoxExpressionFormatter.kt:73) ~[recode-0.1.0-beta.10+1.20.4-LATEST.jar:?]
    at net.minecraft.class_4717.modifyReturnValue$zoi000$recode$highlightAndPreview(class_4717.java:554) ~[client-intermediary.jar:?]
    at net.minecraft.class_4717.method_23931(class_4717.java:321) ~[client-intermediary.jar:?]
    at net.minecraft.class_342.method_48579(class_342.java:404) ~[client-intermediary.jar:?]
    at net.minecraft.class_339.method_25394(class_339.java:66) ~[client-intermediary.jar:?]
    at net.minecraft.class_408.method_25394(class_408.java:192) ~[client-intermediary.jar:?]
    at net.minecraft.class_437.method_47413(class_437.java:110) ~[client-intermediary.jar:?]
-- snip --
```

completely stolen from reason's discord report but I am struggling to cope

--- a/src/main/java/io/github/homchom/recode/ui/text/MiniMessageHighlighter.kt
+++ b/src/main/java/io/github/homchom/recode/ui/text/MiniMessageHighlighter.kt
@@ -1,7 +1,6 @@
 package io.github.homchom.recode.ui.text
 
 import io.github.homchom.recode.util.std.interpolate
-import net.kyori.adventure.text.BuildableComponent
 import net.kyori.adventure.text.Component
 import net.kyori.adventure.text.ComponentLike
 import net.kyori.adventure.text.format.TextDecoration

--- a/src/main/java/io/github/homchom/recode/ui/text/MiniMessageHighlighter.kt
+++ b/src/main/java/io/github/homchom/recode/ui/text/MiniMessageHighlighter.kt
@@ -2,7 +2,6 @@ package io.github.homchom.recode.ui.text
 
 import io.github.homchom.recode.util.std.interpolate
 import net.kyori.adventure.text.Component
-import net.kyori.adventure.text.ComponentLike
 import net.kyori.adventure.text.format.TextDecoration
 import net.kyori.adventure.text.minimessage.Context
 import net.kyori.adventure.text.minimessage.MiniMessage
@@ -59,7 +58,7 @@ class MiniMessageHighlighter(private val standardTags: TagResolver) {
      * - All other valid tags are treated as "literals" and colored gray.
      */
     @Suppress("UnstableApiUsage") // TODO: open issue at adventure github
-    fun highlight(input: String): ComponentLike {
+    fun highlight(input: String): Component {
         // handle the special case of inputs with <reset> parser directives
         // TODO: determine if there is a better way to do this
         val splitByResets = input.split("<reset>")
@@ -69,7 +68,7 @@ class MiniMessageHighlighter(private val standardTags: TagResolver) {
                 literal("<reset>")
             }
             append(highlight(splitByResets.last()))
-        }
+        }.asComponent()
 
         fun TagNode.input() = with(token()) { input.substring(startIndex(), endIndex()) }
         fun ValueNode.input() = with(token()) { input.substring(startIndex(), endIndex()) }

--- a/src/main/java/io/github/homchom/recode/ui/text/MiniMessageHighlighter.kt
+++ b/src/main/java/io/github/homchom/recode/ui/text/MiniMessageHighlighter.kt
@@ -1,6 +1,7 @@
 package io.github.homchom.recode.ui.text
 
 import io.github.homchom.recode.util.std.interpolate
+import net.kyori.adventure.text.BuildableComponent
 import net.kyori.adventure.text.Component
 import net.kyori.adventure.text.ComponentLike
 import net.kyori.adventure.text.format.TextDecoration
@@ -116,7 +117,7 @@ class MiniMessageHighlighter(private val standardTags: TagResolver) {
             instance.deserializeToTree(input)
         } catch (e: Exception) {
             // https://github.com/KyoriPowered/adventure/issues/1011
-            return literalText(input, style().red())
+            return literalText(input, style().red()).asComponent()
         }
         buildNewInput(root)
         return instance.deserialize(newInput.toString())

